android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.fxsuggest
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.withContext
+import mozilla.appservices.suggest.SuggestIngestionConstraints
+import mozilla.appservices.suggest.SuggestStore
+import mozilla.appservices.suggest.Suggestion
+import mozilla.appservices.suggest.SuggestionQuery
+import mozilla.components.support.base.coroutines.Dispatchers.Cached
+
+/**
+ * A coroutine-aware wrapper around the synchronous [SuggestStore] interface.
+ *
+ * @property context The Android application context.
+ */
+class FxSuggestStorage(
+    context: Context,
+) {
+    // Lazily initializes the store on first use. `Context.getDatabasePath()` does I/O, so this
+    // value must be accessed in the read or write scope.
+    internal val store: Lazy<SuggestStore> = lazy {
+        SuggestStore(context.getDatabasePath(DATABASE_NAME).absolutePath)
+    }
+
+    // We expect almost all Suggest storage operations to be reads, with infrequent writes. Our
+    // coroutine scopes reflect this workload: we use the dynamically scaled I/O dispatcher for
+    // reads, and a short-lived thread pool for writes.
+    internal var readScope = CoroutineScope(IO)
+    internal var writeScope = CoroutineScope(Cached)
+
+    /**
+     * Queries the store for suggestions.
+     */
+    suspend fun query(query: SuggestionQuery): List<Suggestion> = withContext(readScope.coroutineContext) {
```

Comment:
`PlacesStorage` goes a step further and redefines the data types, too, but I don't think we need to go that far. I think using these in our public API _does_ mean that `ComponentsDependencies.mozilla_suggest` has to become an `api` dependency, not an `implementation` one, though.

Reply:
I would definitely prefer to not introduce additional types

Reply:
> I think using the Application Services Suggest types in our public API _does_ mean that `ComponentsDependencies.mozilla_suggest` has to become an `api` dependency, not an `implementation` one, though.

This is fine. Other examples in [Places](https://searchfox.org/mozilla-mobile/rev/e45f12112cf8f679691f83ac2dfa2b46919887f3/firefox-android/android-components/components/browser/storage-sync/build.gradle#30).